Source/WebCore/Modules/webcodecs/WebCodecsVideoFrameAlgorithms.cpp

```diff
@@ -216,13 +218,22 @@ ExceptionOr<CombinedPlaneLayout> computeLayoutAndAllocationSize(const DOMRectIni
             computedLayout.destinationOffset = minAllocationSize;
             computedLayout.destinationStride = computedLayout.sourceWidthBytes;
         }
-        // FIXME: validate we do not get over max values.
-        auto planeSize = computedLayout.destinationStride * computedLayout.sourceHeight;
-        auto planeEnd = planeSize + computedLayout.destinationOffset;
 
+        size_t planeSize, planeEnd;
+        if (!WTF::safeMultiply(computedLayout.destinationStride, computedLayout.sourceHeight, planeSize) || planeSize > std::numeric_limits<uint32_t>::max())
+            return Exception { TypeError, "planeSize is too big"_s };
+
+        if (!WTF::safeAdd(planeSize, computedLayout.destinationOffset, planeEnd) || planeEnd > std::numeric_limits<uint32_t>::max())
+            return Exception { TypeError, "planeEnd is too big"_s };
+
+        endOffsets.uncheckedAppend(planeEnd);
         minAllocationSize = std::max(minAllocationSize, planeEnd);
 
-        // FIXME validate endOffsets.
+        for (size_t j = 1; j < i; ++j) {
+            if (planeEnd > computedLayouts[j].destinationOffset && endOffsets[j] > computedLayout.destinationOffset)
+                return Exception { TypeError, "planes are overlapping"_s };
```

Comment:
Do both of these have to be true to fail, or should it fail if either is too big?

Reply:
As per spec, this checks whether the current plane end is after the previous plane start AND the current plane start is before the previous plane end, so overlapping.